### PR TITLE
Support Python 3.8

### PR DESCRIPTION
**Pull request by neiljp**
This PR primarily adds a Python 3.8 test entry in Travis and claims support for installation with it via `setup.py`.

We no longer need special options to test Python 3.7 (or 3.8), so I've removed this too.

**NOTE**: Pending a later release of the `zulip` package, I'm keeping this as a draft for now, since while the tests run fine, the current released zulip API version doesn't support 3.8. I've tested this is the case using a docker image.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -64,10 +64,11 @@ def long_description():
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: Implementation :: CPython',
         'Programming Language :: Python :: Implementation :: PyPy',
     ],
-    python_requires='>=3.5, <3.8',
+    python_requires='>=3.5, <3.9',
```
Comment (rht) [inline setup.py]: I think `<=3.8` is more symmetric in relation to `>=3.5`.

Comment (neiljp) [reply setup.py]: That seems reasonable.

Still pending on a release of `zulip` for now, though.

**Comment by LeifW** [comment]
Trying to run on python 3.8, after I login, it crashes with:
```
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): %20https:443
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): inner-product.zulipchat.com:443
DEBUG:urllib3.connectionpool:https://inner-product.zulipchat.com:443 "POST /api/v1/fetch_api_key HTTP/1.1" 200 None
INFO:root:

module 'platform' has no attribute 'linux_distribution'


ERROR:root:module 'platform' has no attribute 'linux_distribution'
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/zulipterminal/cli/run.py", line 241, in main
    Controller(zuliprc_path,
  File "/usr/lib/python3.8/site-packages/zulipterminal/core.py", line 40, in __init__
    self.model = Model(self)
  File "/usr/lib/python3.8/site-packages/zulipterminal/model.py", line 109, in __init__
    self._update_initial_data()
  File "/usr/lib/python3.8/site-packages/zulipterminal/model.py", line 440, in _update_initial_data
    results = {
  File "/usr/lib/python3.8/site-packages/zulipterminal/model.py", line 441, in <dictcomp>
    name: self.exception_safe_result(future)
  File "/usr/lib/python3.8/site-packages/zulipterminal/model.py", line 386, in exception_safe_result
    return future.result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/lib/python3.8/site-packages/zulipterminal/model.py", line 353, in get_messages
    response = self.client.get_messages(message_filters=request)
  File "/usr/lib/python3.8/site-packages/zulip/__init__.py", line 703, in get_messages
    return self.call_endpoint(
  File "/usr/lib/python3.8/site-packages/zulip/__init__.py", line 623, in call_endpoint
    return self.do_api_query(marshalled_request, versioned_url, method=method,
  File "/usr/lib/python3.8/site-packages/zulip/__init__.py", line 505, in do_api_query
    self.ensure_session()
  File "/usr/lib/python3.8/site-packages/zulip/__init__.py", line 450, in ensure_session
    session.headers.update({"User-agent": self.get_user_agent()})
  File "/usr/lib/python3.8/site-packages/zulip/__init__.py", line 466, in get_user_agent
    vendor, vendor_version, dummy = platform.linux_distribution()
AttributeError: module 'platform' has no attribute 'linux_distribution'
```
This says that's been removed in Python 3.8: https://github.com/pymssql/pymssql/issues/586

**Comment by neiljp** [comment]
@LeifW The issue you are seeing is within the `zulip` library; there is a fix in that repo and a release of that is pending right now.

**Comment by LeifW** [comment]
Found https://github.com/zulip/python-zulip-api/issues/530

File: setup.py (hunk 1 above)
Comment (neiljp) [reply setup.py]: Fixed in current version.

**Comment by neiljp** [comment]
After a brief confusion over whether `pipenv` was breaking things, I'm merging this now so that we can get things tested - it seems OK in CI :+1:

**Comment by neiljp** [comment]
@LeifW Please let us know if the latest development version has any 3.8-related issues for you here, or in a new issue :)

**Pull request merged**